functions/Update-DbaInstance.ps1

```diff
@@ -289,13 +300,96 @@ function Update-DbaInstance {
+                [ValidateNotNullOrEmpty()]
+                [string]$Path
+
+            )
+            if ($Path.StartsWith("\\")) {
+                return $filepath
+            }
+
+            $servername = $ComputerName.ComputerName
+            $newpath = Join-Path "\\$servername\" $Path.replace(':', '$')
+            return $newpath
+        }
+        function Copy-UncFile {
+            <#
+
+                SYNOPSIS
+                Internal function. Uses PSDrive to copy file to the remote system.
+
+                #>
+            param (
+                [Parameter(Mandatory)]
+                [ValidateNotNullOrEmpty()]
+                [DbaInstanceParameter]$ComputerName,
+
+                [Parameter(Mandatory)]
+                [ValidateNotNullOrEmpty()]
+                [string]$Path,
+
+                [Parameter(Mandatory)]
+                [ValidateNotNullOrEmpty()]
+                [string]$Destination,
+
+                [PSCredential]$Credential
+            )
+            if (([DbaInstanceParameter]$groupItem.ComputerName).IsLocalHost) {
+                $remoteFolder = $Destination
+            } else {
+                $uncFileName = Join-AdminUnc -ComputerName $ComputerName -Path $Destination
+                $driveSplat = @{
+                    Name       = 'UpdateCopy'
+                    Root       = $uncFileName
+                    PSProvider = 'FileSystem'
+                }
+                if ($Credential) { $driveSplat.Credential = $Credential }
+                $null = New-PSDrive @driveSplat -ErrorAction Stop
+                $remoteFolder = 'UpdateCopy:\'
+            }
+            try {
+                Copy-Item -Path $Path -Destination $remoteFolder -ErrorAction Stop
+            } finally {
+                if (-Not ([DbaInstanceParameter]$groupItem.ComputerName).IsLocalHost) {
+                    $null = Remove-PSDrive -Name UpdateCopy -Force
+                }
+            }
+        }
+        function Test-NetworkPath {
+            <#
+
+            SYNOPSIS
+            Internal function. Tests if a path is a network path
+
+            #>
+            param (
+                [Parameter(ValueFromPipeline)]
+                [string]$Path
+            )
+            begin { $pathList = @() }
+            process { $pathList += $Path -like '\\*' }
+            end { return $pathList -contains $true }
+        }
     }
+
     process {
         if (Test-FunctionInterrupt) { return }
 
         #Resolve all the provided names
         $resolvedComputers = @()
-        $pathIsNetwork = $Path | ForEach-Object -Begin { $o = @() } -Process { $o += $_ -like '\\*' } -End { $o -contains $true }
+        $pathIsNetwork = $Path | Test-NetworkPath
         foreach ($computer in $ComputerName) {
             $null = Test-ElevationRequirement -ComputerName $computer -Continue
             if (!$computer.IsLocalHost -and -not $notifiedCredentials -and -not $Credential -and $pathIsNetwork) {
```

Comment:
If you would not mind also updating the `!` to be `-not` as well for consistency 😉

Reply:
fixed